### ISSUE TITLE
Cannot boot any ROM with internal firmware

### ISSUE THREAD

**Issue by Brankale** [Arisotura/melonDS#1335]
the problem was introduced with "custom path support" e665e25. When you try to boot any game the emulator just shows a blank screen and the fps counter.

**Comment by Arisotura** [comment]
I don't have this problem here

**Comment by iamthelaw2041** [comment]
@Brankale What specific games have you tried?

**Comment by Brankale** [comment]
Can be related in some way to the melonDS.ini file? Because I started from scratch.

Additional info:
- Tested with Windows 7 and 10
- Tested with and without a dedicated GPU, so it's not a driver fault
- Tested with few pokemon & Professor Layton games
- Downloading the .exe  from the previous commit, everything works fine

**Comment by iamthelaw2041** [comment]
Re-downloaded and started from scratch. All settings default. Pokemon Pearl works fine. Tested with and without external BIOS. I'm on Windows 7. 

![Capture](https://user-images.githubusercontent.com/94656803/148581221-cf3633a5-8492-429a-a2a3-8fdbda8bcfe0.PNG)

**Comment by Brankale** [comment]
I've tried to delete save files, to change the name of the rom and I've tried it also inside Windows Sandbox but I always get a blank screen. I can't figure out what can be the problem on my side.

**Comment by Arisotura** [comment]
can you send your melonDS.ini file?

**Comment by nadiaholmquist** [comment]
There's something odd going on here, not sure if this is the same issue but it may be related.

If I load a game with direct boot enabled (using Open ROM or the recent list), the first game loaded will work but trying to load any game after that will result in nothing happening... Except with Tetris DS?

Happens regardless of using external BIOS or not.

**Comment by Brankale** [comment]
melonDS.ini file [melonDS.zip](https://github.com/Arisotura/melonDS/files/7830393/melonDS.zip)

**Comment by nadiaholmquist** [comment]
This is what is printed in the console when loading a game and nothing happens.
```
Resetting JIT block cache...
done resetting jit mem
done resetting jit mem
remapping SWRAM
MAC: 00:09:BF:11:22:33
FW: WIFI CRC16 = GOOD
FW: AP1 CRC16 = GOOD
FW: AP2 CRC16 = GOOD
FW: AP3 CRC16 = GOOD
FW: USER0 CRC16 = GOOD
FW: USER1 CRC16 = BAD
Game code: YR9E
ROM entry: 04000000 00000002
Cart ID: 00003FC2
Re-encrypting cart secure area
remapping SWRAM
Secure area decryption OK
remapping DTCM 0 1000 ffffffff ffffffff
PU: region 0 = 04000033 : enabled, 04000000-08000000
PU: region 0 = 04000033 : enabled, 04000000-08000000
PU: region 1 = 0200002B : enabled, 02000000-02400000
PU: region 1 = 0200002B : enabled, 02000000-02400000
PU: region 2 = 00000000 : disabled, 00000000-00000002
PU: region 2 = 00000000 : disabled, 00000000-00000002
PU: region 3 = 08000035 : enabled, 08000000-10000000
PU: region 3 = 08000035 : enabled, 08000000-10000000
PU: region 4 = 0300001B : enabled, 03000000-03004000
PU: region 4 = 0300001B : enabled, 03000000-03004000
PU: region 5 = 00000000 : disabled, 00000000-00000002
PU: region 5 = 00000000 : disabled, 00000000-00000002
PU: region 6 = FFFF001D : enabled, FFFF0000-FFFF8000
PU: region 6 = FFFF001D : enabled, FFFF0000-FFFF8000
PU: region 7 = 027FF017 : enabled, 027FF000-02800000
PU: region 7 = 027FF017 : enabled, 027FF000-02800000
remapping DTCM 3000000 3004000 0 fffff000
Game is now booting
remapping DTCM ffffffff ffffffff 3000000 2ffc000
PU: region 0 = 04000033 : enabled, 04000000-08000000
PU: region 1 = 0200002D : enabled, 02000000-02800000
PU: region 2 = 027E0021 : enabled, 027E0000-02800000
PU: region 3 = 08000035 : enabled, 08000000-10000000
PU: region 4 = 027E001B : enabled, 027E0000-027E4000
PU: region 5 = 0100002F : enabled, 01000000-02000000
PU: region 6 = FFFF001D : enabled, FFFF0000-FFFF8000
PU: region 7 = 027FF017 : enabled, 027FF000-02800000
remapping DTCM 27e0000 27e4000 ffffffff ffffffff
PU region 0: 04000000-08000000, user=00 priv=07
15111011/05100011
PU region 1: 02000000-02800000, user=70 priv=77
15111011/05100011
PU region 2: 027E0000-02800000, user=00 priv=00
15111011/05100011
PU region 3: 08000000-10000000, user=00 priv=03
15111011/05100011
PU region 4: 027E0000-027E4000, user=00 priv=03
15111011/05100011
PU region 5: 01000000-02000000, user=00 priv=07
15111011/05100011
PU region 6: FFFF0000-FFFF8000, user=50 priv=55
15111011/05100011
PU region 7: 027FF000-02800000, user=00 priv=03
15111011/05100011
PU: region 1 = 0200002B : enabled, 02000000-02400000
PU region 0: 04000000-08000000, user=00 priv=07
15111011/05100011
PU region 1: 02000000-02400000, user=70 priv=77
15111011/05100011
PU region 2: 027E0000-02800000, user=00 priv=00
15111011/05100011
PU region 3: 08000000-10000000, user=00 priv=03
15111011/05100011
PU region 4: 027E0000-027E4000, user=00 priv=03
15111011/05100011
PU region 5: 01000000-02000000, user=00 priv=07
15111011/05100011
PU region 6: FFFF0000-FFFF8000, user=50 priv=55
15111011/05100011
PU region 7: 027FF000-02800000, user=00 priv=03
15111011/05100011
PU: region 2 = 023E0021 : enabled, 023E0000-02400000
PU region 0: 04000000-08000000, user=00 priv=07
15111011/05100011
PU region 1: 02000000-02400000, user=70 priv=77
15111011/05100011
PU region 2: 023E0000-02400000, user=00 priv=00
15111011/05100011
PU region 3: 08000000-10000000, user=00 priv=03
15111011/05100011
PU region 4: 027E0000-027E4000, user=00 priv=03
15111011/05100011
PU region 5: 01000000-02000000, user=00 priv=07
15111011/05100011
PU region 6: FFFF0000-FFFF8000, user=50 priv=55
15111011/05100011
PU region 7: 027FF000-02800000, user=00 priv=03
15111011/05100011
PU region 3: 08000000-10000000, user=00 priv=01
15115011/05100011
WIFI OFF
unknown ARM9 IO write32 04001060 00000000 01FF8580
unknown ARM9 IO write32 04001064 00000000 01FF8580
unknown ARM9 IO write32 04001068 00000000 01FF8580
unknown arm7 write16 00000000 FFFF @ 037FDF34
```

**Comment by Brankale** [comment]
> If I load a game with direct boot enabled (using Open ROM or the recent list), the first game loaded will work but trying to load any game after that will result in nothing happening. Happens regardless of using external BIOS or not.

I can confirm this issue

**Comment by iamthelaw2041** [comment]
Wonder why I'm not getting this problem.

**Comment by Brankale** [comment]
These are the logs pre/post commit: [logs.zip](https://github.com/Arisotura/melonDS/files/7831810/logs.zip)

The only differences are:
- `Secure area decryption failed` instead of OK
- `Save file: rom_path` missing
- `undefined THUMB9 instruction E7FFDEFF @ 020006E0`

**Comment by Arisotura** [comment]
are you using encrypted ROMs? you'll need --

oh wait I think I know what the issue is.

blarg.

**Comment by Brankale** [comment]
Both issues are now solved with 80564ba947eae83bdcb4f18ece7e36617a217fbb on my side. If also @nadiaholmquist doesn't have any problem we can close this issue.
